sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/placement/MarathonConstraintParser.java

```
@@ -285,10 +304,13 @@ public PlacementRule run(StringMatcher taskFilter, String fieldName, String oper
                         "Unable to parse max parameter as integer for '%s' operation: %s",
                         operatorName, parameter), e);
             }
-            if (isHostname(fieldName)) {
-                return new RoundRobinByHostnameRule(num, taskFilter);
-            } else {
-                return new RoundRobinByAttributeRule(fieldName, num, taskFilter);
+
+            switch (PlacementUtils.getField(fieldName)) {
```

should the default case throw an `UnsupportedOperationException`?